File: socketd-broker/socketd-broker-websocket/src/main/java/org/noear/socketd/broker/websocket/WsClient.java
package org.noear.socketd.broker.websocket;

import org.noear.socketd.client.*;
import org.noear.socketd.protocol.Channel;
import org.noear.socketd.protocol.Session;
import org.noear.socketd.protocol.impl.SessionDefault;

import java.io.IOException;
import java.util.concurrent.TimeoutException;

/**
 * @author noear
 * @since 2.0
 */
public class WsClient extends ClientBase<WsExchanger> {

    public WsClient(ClientConfig clientConfig) {
        super(clientConfig, new WsExchanger());
    }

    @Override
    public Session open() throws IOException {
        ClientConnector connector = new WsClientConnector(this);
        Channel channel = new ClientChannel(connector.connect(), connector);
        return new SessionDefault(channel);
    }
}

File: javascript/socketd-examples/demo01-js-ws-esm/js/socketd/transport_websocket/WsClient.js
import { ClientBase } from "../transport/client/Client";
import { WsChannelAssistant } from "./WsChannelAssistant";
import { WsClientConnector } from "./WsClientConnector";
export class WsClient extends ClientBase {
    constructor(clientConfig) {
        super(clientConfig, new WsChannelAssistant(clientConfig));
    }
    createConnector() {
        return new WsClientConnector(this);
    }
}
